From 82912151f38f9a0318ca6edb014a98132286e543 Mon Sep 17 00:00:00 2001
From: Paolo Bonzini <pbonzini@redhat.com>
Date: Fri, 28 Aug 2026 16:14:56 +0200
Subject: [PATCH] d: pass link arguments through unix_args_to_native

Commit decc3e assumed the sanity check goes through CompilerArgs,
but it does not.  Add to _sanity_check_compile_args() the
missing call to DCompiler.unix_args_to_native().

Fixes: #16115
Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>
---
 mesonbuild/compilers/d.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/mesonbuild/compilers/d.py b/mesonbuild/compilers/d.py
index 367355828738..6cab9e8fc45f 100644
--- a/mesonbuild/compilers/d.py
+++ b/mesonbuild/compilers/d.py
@@ -402,6 +402,7 @@ def _sanity_check_source_code(self) -> str:
     def _sanity_check_compile_args(self, sourcename: str, binname: str
                                    ) -> T.Tuple[T.List[str], T.List[str]]:
         args, largs = super()._sanity_check_compile_args(sourcename, binname)
+        largs = self.unix_args_to_native(largs)
         largs.extend(self._get_target_arch_args())
         return args, largs
 
